### PR TITLE
remove mintInitialSupply from basic-behaviors.js

### PR DESCRIPTION
**Pull request by warner**
fix(vats): remove mintInitialSupply from basic behaviors

basic-behaviors.js included a function named `mintInitialSupply()`,
which (if run) would create the `centralSupply` vat, which can create
an initial supply of IST tokens. We obviously don't enable this in the
production config, but several unit tests and testnet configs need it,
as those environments have no other source of tokens.

To assist with auditing, this commit removes `mintInitialSupply` from
the basic behaviors entirely, so the code won't even be present in
production.

The code was moved to `demoIssuers.js`, which is used by
`sim-behaviors.js`, which is in turn used by the sim-chain and a few
tests.

refs #6687

### CODE REVIEW

**Comment by warner** [comment]
might supercede #7160

File: packages/vats/decentral-test-vaults-config.json [hunk 1]
```diff
@@ -169,9 +169,6 @@
     "bridge": {
       "sourceSpec": "@agoric/vats/src/vat-bridge.js"
     },
-    "centralSupply": {
-      "sourceSpec": "@agoric/vats/src/centralSupply.js"
```
Comment (dckc) [inline packages/vats/decentral-test-vaults-config.json:173]: :dart:

File: packages/vats/src/core/basic-behaviors.js [hunk 2]
```diff
@@ -440,11 +439,10 @@ harden(makeClientBanks);
 export const installBootContracts = async ({
   consume: { vatAdminSvc, zoe },
   installation: {
-    produce: { centralSupply, mintHolder },
```
Comment (dckc) [inline packages/vats/src/core/basic-behaviors.js:443]: :dart:

File: packages/vats/src/core/basic-behaviors.js [hunk 3]
```diff
@@ -456,46 +454,7 @@ export const installBootContracts = async ({
 };
 
 /**
- * Mint IST genesis supply.
- *
- * @param { BootstrapPowers & {
- *   vatParameters: { argv: { bootMsg?: typeof bootMsgEx }},
- * }} powers
- */
-export const mintInitialSupply = async ({
```
Comment (dckc) [inline packages/vats/src/core/basic-behaviors.js:465]: :wave:

**Review by dckc — APPROVED** [review]
thanks! looks good.

File: packages/vats/decentral-test-vaults-config.json (hunk 1 above)
Comment (dckc) [reply packages/vats/decentral-test-vaults-config.json:173]: oops! it needs to be gone from the devnet config too:

```
decentral-core-config.json:    "centralSupply": {
decentral-core-config.json:      "sourceSpec": "@agoric/vats/src/centralSupply.js"
decentral-demo-config.json:    "centralSupply": {
decentral-demo-config.json:      "sourceSpec": "@agoric/vats/src/centralSupply.js"
decentral-devnet-config.json:    "centralSupply": {
decentral-devnet-config.json:      "sourceSpec": "@agoric/vats/src/centralSupply.js"
```

ideally core too, but that's not critical

**Review by dckc — CHANGES_REQUESTED** [review]
oops! needs to be gone from devnet config

**Review by dckc — APPROVED** [review]
good stuff.

File: packages/vats/decentral-test-vaults-config.json (hunk 1 above)
Comment (warner) [reply packages/vats/decentral-test-vaults-config.json:173]: removed from core-config and devnet-config, but left in demo-config

**Comment by warner** [comment]
I think this PR might be a mistake. Reverting to draft until I sort out my confusion.

**Comment by warner** [comment]
@mfig confirmed that this is a bad idea. abandoning.

**Pull request closed without merging**